src/instana/instrumentation/pep0249.py

```diff
@@ -2,139 +2,169 @@
             except Exception as e:
-                if scope.span:
-                    scope.span.log_exception(e)
+                if span:
+                    span.record_exception(e)
                 raise
             else:
                 return result
 
-    def callproc(self, proc_name, params):
+    def callproc(self, proc_name, params) -> None:
         tracer, parent_span, operation_name = get_tracer_tuple()
 
         # If not tracing or we're being called from sqlalchemy, just pass through
-        if (tracing_is_off() or (operation_name == "sqlalchemy")):
+        if tracing_is_off() or (operation_name == "sqlalchemy"):
             return self.__wrapped__.execute(proc_name, params)
 
-        with tracer.start_active_span(self._module_name, child_of=parent_span) as scope:
+        parent_context = parent_span.get_span_context() if parent_span else None
+        with tracer.start_as_current_span(
+            self._module_name, span_context=parent_context
+        ) as span:
             try:
-                self._collect_kvs(scope.span, proc_name)
-
+                self._collect_kvs(span, proc_name)
                 result = self.__wrapped__.callproc(proc_name, params)
-            except Exception as e:
-                if scope.span:
-                    scope.span.log_exception(e)
-                raise
+            except Exception:
+                try:
+                    result = self.__wrapped__.execute(proc_name, params)
+                except Exception as e_execute:
+                    if span:
+                        span.record_exception(e_execute)
+                    raise
+                else:
+                    return result
             else:
                 return result
 
 
 class ConnectionWrapper(wrapt.ObjectProxy):
-    __slots__ = ('_module_name', '_connect_params')
+    __slots__ = ("_module_name", "_connect_params")
 
-    def __init__(self, connection, module_name, connect_params):
+    def __init__(self, connection, module_name, connect_params) -> None:
         super(ConnectionWrapper, self).__init__(wrapped=connection)
         self._module_name = module_name
         self._connect_params = connect_params
 
     def __enter__(self):
         return self
 
-    def cursor(self, *args, **kwargs):
+    def cursor(self, *args, **kwargs) -> CursorWrapper:
         return CursorWrapper(
             cursor=self.__wrapped__.cursor(*args, **kwargs),
             module_name=self._module_name,
             connect_params=self._connect_params,
-            cursor_params=(args, kwargs) if args or kwargs else None)
+            cursor_params=(args, kwargs) if args or kwargs else None,
+        )
 
-    def begin(self):
-        return self.__wrapped__.begin()
+    def close(self) -> None:
+        return self.__wrapped__.close()
 
-    def commit(self):
+    def commit(self) -> None:
         return self.__wrapped__.commit()
 
-    def rollback(self):
+    def rollback(self) -> None:
```

Comment:
The return type hint must be corrected.

Reply:
rollback returns None

![image](https://github.com/user-attachments/assets/4d4350a8-7b66-4927-9852-a3f22a4f0e8f)